src/commands/import.js

```
@@ -0,0 +1,136 @@
+    '$ radar import -f sarif -e warning,note scan.sarif ' + '(treat lower severities as errors)'.grey
+  ],
+  run: async (toolbox, args) => {
+    const { log, scanners: availableScanners, telemetry } = toolbox
+
+    // Set defaults for args and options.
+    args.FORMAT ??= 'security'
+
+    // Normalize and/or rewrite args and options.
+    args.INPUT = path.resolve(path.normalize(args.INPUT))
+
+    // Validate args and options.
+    if (args.FORMAT !== 'sarif' && args.FORMAT !== 'security') throw new Error('FORMAT must be one of \'sarif\' or \'security\'')
+    if (args.ESCALATE) args.ESCALATE.split(',').map(severity => {
+      if (args.FORMAT === 'security' && severity !== 'moderate' && severity !== 'low') throw new Error(`Severity to escalate must be 'moderate' or 'low'`)
+      if (args.FORMAT === 'sarif' && severity !== 'warning' && severity !== 'note') throw new Error(`Severity to escalate must be 'warning' or 'note'`)
+    })
+
+    // Derive scan parameters.
+    const escalations = args.ESCALATE?.split(',').map(severity => {
+      if (severity === 'moderate') return 'warning'
+      if (severity === 'low') return 'note'
+      return severity
+    })
+
+    // Check that telemetry is enabled.
+    if (!args.QUIET && !telemetry.enabled) {
+      log(`ERROR: Telemetry not enabled.`)
+      log(`Terminating with exit code 16. See 'radar help import' for list of possible exit codes.`)
+      return 16 // exit code
+    }
+
+    // Results include the log and the SARIF findings.
+    const results = { log: `Import from "${args.INPUT}"` }
+    results.sarif = JSON.parse(fs.readFileSync(args.INPUT, 'utf8'))
+
+    // Read scanner names from the input SARIF.
+    const scanners = []
+    for (const run of results.sarif.runs) {
+      const scanner = run.tool.driver?.properties?.scanner_name ?? run.tool.driver.name
+      scanners.push(scanner)
+    }
+
+    // Check for unsupported scanners.
+    try {
```

What if a SARIF has an unknown scanner, but also a supported scanner? Should it fail to import if the SARIF has any unknown scanners at all?